airflow/contrib/hooks/databricks_hook.py

```diff
@@ -0,0 +1,201 @@
+            try:
+                response = request_func(
+                    url,
+                    json=json,
+                    auth=auth,
+                    headers=USER_AGENT_HEADER,
+                    timeout=self.timeout_seconds,
+                    verify=False)
+                if response.status_code == requests.codes.ok:
+                    return response.json()
+                else:
+                    # In this case, the user probably made a mistake.
+                    # Don't retry.
+                    raise AirflowException('Response: {0}, Status Code: {1}'.format(
+                        response.content, response.status_code))
+            except (requests_exceptions.ConnectionError,
+                    requests_exceptions.Timeout) as e:
+                logging.error(('Attempt {0} API Request to Databricks failed ' +
+                              'with reason: {1}').format(attempt_num, e))
+        raise AirflowException(('API requests to Databricks failed {} times. ' +
+                               'Giving up.').format(self.retry_limit))
+
+    def submit_run(self, json):
+        """
+        Utility function to call the ``api/2.0/jobs/runs/submit`` endpoint.
+
+        :param json: The data used in the body of the request to the ``submit`` endpoint.
+        :type json: dict
+        :return: the run_id as a string
+        :rtype: string
+        """
+        response = self._do_api_call(SUBMIT_RUN_ENDPOINT, json)
+        return response['run_id']
+
+    def get_run_page_url(self, run_id):
+        json = {'run_id': run_id}
+        response = self._do_api_call(GET_RUN_ENDPOINT, json)
+        return response['run_page_url']
+
+    def get_run_state(self, run_id):
+        json = {'run_id': run_id}
+        response = self._do_api_call(GET_RUN_ENDPOINT, json)
+        state = response['state']
+        life_cycle_state = state['life_cycle_state']
+        # result_state may not be in the state if not terminal
+        result_state = state.get('result_state', None)
+        state_message = state['state_message']
+        return RunState(life_cycle_state, result_state, state_message)
+
+    def cancel_run(self, run_id):
+        json = {'run_id': run_id}
+        self._do_api_call(CANCEL_RUN_ENDPOINT, json)
+
+
+RUN_LIFE_CYCLE_STATES = [
+    'PENDING',
+    'RUNNING',
+    'TERMINATING',
+    'TERMINATED',
+    'SKIPPED',
+    'INTERNAL_ERROR'
+]
+
+
+class RunState:
+    """
+    Utility class for the run state concept of Databricks runs.
+    """
+    def __init__(self, life_cycle_state, result_state, state_message):
+        self.life_cycle_state = life_cycle_state
+        self.result_state = result_state
+        self.state_message = state_message
+
+    @property
+    def is_terminal(self):
+        if self.life_cycle_state not in RUN_LIFE_CYCLE_STATES:
+            raise AirflowException(('Unexpected life cycle state: {}: If the state has ' +
```

Comment:
Also a nit, but the `+` are superfluous here, since python will concatenate those strings automatically.